modules/images/webp-uploads/load.php

```diff
@@ -30,12 +30,21 @@
 function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
 	// This should take place only on the JPEG image.
 	$valid_mime_transforms = webp_uploads_get_supported_image_mime_transforms();
+
 	// Not a supported mime type to create the sources property.
 	$mime_type = get_post_mime_type( $attachment_id );
 	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
 		return $metadata;
 	}
 
+	// Don't do anything if only the original mime type is needed.
+	if (
+		1 === count( $valid_mime_transforms[ $mime_type ] ) &&
+		current( $valid_mime_transforms[ $mime_type ] ) === $mime_type
+	) {
+		return $metadata;
+	}
+
```

Comment:
I think this needs to be removed. While I can see how it would be nice to not run the logic below, it also populates the `sources` data even for the image formats that already exist, and we need to make sure that runs for consistent behavior.

As far as I can tell, this function is already set up correctly to not double-create any image in a MIME type that already exists, so this check here isn't necessary from that perspective either.

Maybe the first sub-bullet point in https://github.com/WordPress/performance/issues/187#issuecomment-1061277586 was confusing in that regard, my intention there was just to clarify this needs to be the case, not that we actually need to add code for this if it's already working as expected (which I think it does).